### PR TITLE
Make it possible to give --libdir an absolute path

### LINKED ISSUE #5398: OpenSSL 1.1. does not honor --libdir
I asked for the libraries to be placed in `/opt/local/test/b/lib`:

```
$ ./config --openssldir=/opt/local/test/a/bin --prefix=/opt/local/test  --libdir=/opt/local/test/b/lib
```

Here is where they showed up:

```
$ find /opt/local/test -name 'libssl*'
/opt/local/test/opt/local/test/b/lib/libssl.so.1.1
/opt/local/test/opt/local/test/b/lib/libssl.a
/opt/local/test/opt/local/test/b/lib/libssl.so
/opt/local/test/opt/local/test/b/lib/pkgconfig/libssl.pc
```

No environmental variables are set that could interfere with OpenSSL scripts, like `INSTALL_PREFIX`.

### PR DESCRIPTION
**Pull request by levitte**
With this, we introduce the make variable 'libdir', which differs from
'LIBDIR' not only in casing, but also by being the absolute path to
the library installation directory.  This variable is intentionally
compatible with the GNU coding standards.

When --libdir is given an absolute path, it is considered as a value
according to GNU coding standards, and the variables LIBDIR and libdir
will be this:

    LIBDIR=
    libdir=/absolute/path

When --libdir is given a relative path (just the name of the desired
library directory), or not given at all, it is considered as a
"traditional" OpenSSL value, and the variables LIBDIR and libdir will
be this:

    LIBDIR=relativepath
    libdir=$(INSTALLTOP)/$(LIBDIR)

Fixes #5398

### CODE REVIEW

**Comment by levitte** [comment]
@noloader, I think you're interested in this.  Would you mind testing this?

**Review by richsalz — APPROVED** [review]
Perhaps wait for @noloader's feedback, but this LGTM.

**Comment by levitte** [comment]
Merged.  e454f3add638fda5c2aa32cd368c8929c0b1eb09

(@noloader, you will simply have to test this on the next pre-release, due tomorrow)

**Pull request closed without merging**
